### ISSUE TITLE
GUI board/stones looks bad: pixelisated

### ISSUE THREAD

**Issue by garry-ut99** [Joker2770/qpiskvork#6]
There has been discovered some bitmap rendering issue compared to Piskvork: 
https://github.com/MaciejKozarzewski/AlphaGomoku/issues/2#issuecomment-1804826531
https://github.com/MaciejKozarzewski/AlphaGomoku/issues/2#issuecomment-1805399684
(also ping @nguyencongminh090)

**Comment by Joker2770** [comment]
I think it is OK for the render. Please share by pulling requests if you have better bitmap.

**Comment by garry-ut99** [comment]
I don't think it's matter of "better bitmap".
What I mean is that QPiskvork uses some bitmaps from Piskvork, 
in Piskvork there is not pixelisation and looks much better, 
while the same bitmaps looks worse/pixelisated in QPiskvork.

**Comment by Joker2770** [comment]
@garry-ut99 Well, where do you think the problem is? Could you please fix it?

**Comment by garry-ut99** [comment]
See the same skin HGarden2 looks good in piskvork :

![pisvkork](https://github.com/Joker2770/qpiskvork/assets/72945564/5862e1dc-7464-4934-b247-ac32a5e22d46)

> Well, where do you think the problem is? Could you please fix it?

I don't have an idea, I think maybe @nguyencongminh090 can help you.

**Comment by nguyencongminh090** [comment]
I think use solution from Piskvork is okay. I think  Piskvork slice/crop image and arrange it only (directly in loading). I didn't take a look at QPiskvork code. But from folder, I can't find bitmap file so I think you embed it already. Perhaps, I think you had processed the bitmap image and classify it. Not sure how you did but it seems a bug appeared during process. 

Have you ever taken a look at Pisvork's bitmap size? The key is WIDTH of it. The structure of it is SQUARE and 1 column consist information about background color. You can use some program such as PAINT from Window, zoom at maximum and turn on grid, then look at the last column you will see  a magic.

If you knew it already then the problem is at somewhere else. Also the idea of zoom image is bad. Bitmap image is very small just about 32px. You may know if zoom a small image with low resolution, it will be blur. If you want to make it larger, the only solution is change bitmap. You can try to prepare/ or draw few image and set a size for it then let user choose them.

Unfortunately, I don't have time for coding. My message is just for reference only.

**Comment by garry-ut99** [comment]
Yes, I was thinking the same - that he just upscaled the bitmaps somehow (accidentaly or not).
Upscaling bitmaps is not a good idea, because it reduces their quality, creates blur and pixels.

Better soultion is :
- to use bigger bitmap, and then downscale it when changing grid size, 
because downscaling doesn't reduce quality so much as upscaling
(the bigger bitmap should be created from scratch!)
- or to use 2 or 3 bitmaps for each skin,
 for example: HGArden2_small.bmp, HGarden2_medium.bmp, HGarden2_big.bmp
- or use vector graphics (svg) instead of bitmaps, they are designed to scale up without quality loss,
there is a good tool to convert bitmaps to vectors: https://vanceai.com/posts/vector-magic/

**Comment by Joker2770** [comment]
So sorry for that I have not got better solution, and it is necessary to upscale bitmaps for the global layout currently.
